### PR TITLE
Refactors Contractor

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Some (a lot) of contractor code is old garbage that i should've fixed earlier

## How This Contributes To The Skyrat Roleplay Experience
Better code is better

## Changelog

:cl:
refactor: Refactored contractor
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/contractor/code/datums/contract.dm [hunk 1]
```diff
@@ -183,26 +182,37 @@
 		return
 	// Heal them up - gets them out of crit/soft crit. If omnizine is removed in the future, this needs to be replaced with a
 	// method of healing them, consequence free, to a reasonable amount of health.
+	victim_stage_one(target)
+
+/datum/syndicate_contract/proc/victim_stage_one(mob/living/target)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/contractor/code/datums/contract.dm:188]: Just add a tiny bit of doc to these, please.

File: modular_skyrat/modules/contractor/code/datums/contract.dm [hunk 2]
```diff
@@ -183,26 +182,37 @@
...
 	target.reagents.add_reagent(/datum/reagent/medicine/omnizine, 20)
 
 	target.flash_act()
 	target.adjust_timed_status_effect(10 SECONDS, /datum/status_effect/confusion)
 	target.blur_eyes(5)
 	to_chat(target, span_warning("You feel strange..."))
-	sleep(6 SECONDS)
+	addtimer(CALLBACK(src, .proc/victim_stage_two, target), 6 SECONDS)
+
+/datum/syndicate_contract/proc/victim_stage_two(mob/living/target)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/contractor/code/datums/contract.dm:198]: Ditto.

File: modular_skyrat/modules/contractor/code/datums/contract.dm [hunk 3]
```diff
@@ -183,26 +182,37 @@
...
 	to_chat(target, span_warning("That pod did something to you..."))
 	target.set_timed_status_effect(70 SECONDS, /datum/status_effect/dizziness)
-	sleep(6 SECONDS)
+	addtimer(CALLBACK(src, .proc/victim_stage_three, target), 6 SECONDS)
+
+/datum/syndicate_contract/proc/victim_stage_three(mob/living/target)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/contractor/code/datums/contract.dm:204]: Ditto.

File: modular_skyrat/modules/contractor/code/datums/contract.dm [hunk 4]
```diff
@@ -183,26 +182,37 @@
...
 	to_chat(target, span_warning("Your head pounds... It feels like it's going to burst out your skull!"))
 	target.flash_act()
 	target.adjust_timed_status_effect(20 SECONDS, /datum/status_effect/confusion)
 	target.blur_eyes(3)
-	sleep(3 SECONDS)
+	addtimer(CALLBACK(src, .proc/victim_stage_four, target), 3 SECONDS)
+
+/datum/syndicate_contract/proc/victim_stage_four(mob/living/target)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/contractor/code/datums/contract.dm:212]: Ditto.

File: modular_skyrat/modules/contractor/code/datums/contract.dm [hunk 5]
```diff
@@ -183,26 +182,37 @@
...
 	to_chat(target, span_warning("Your head pounds..."))
-	sleep(10 SECONDS)
+	addtimer(CALLBACK(src, .proc/victim_stage_five, target), 10 SECONDS)
+
+/datum/syndicate_contract/proc/victim_stage_five(mob/living/target)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/contractor/code/datums/contract.dm:217]: Ditto.

File: modular_skyrat/modules/contractor/code/datums/midround/outfit.dm [hunk 6]
```diff
@@ -35,7 +35,6 @@
 	uniform = /obj/item/clothing/under/syndicate
 	glasses = /obj/item/clothing/glasses/night
 
-/datum/outfit/contractor_preview/post_equip(mob/living/carbon/human/guy, visualsOnly)
-    var/obj/item/mod/module/armor_booster/booster = locate() in guy.back
-    booster.active = TRUE
-    guy.update_inv_back()
+/datum/outfit/contractor/upgraded
+	name = "Syndicate Contractor (Upgraded)"
+	back = /obj/item/mod/control/pre_equipped/contractor/upgraded/adminbus
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/contractor/code/datums/midround/outfit.dm:40]: Why's that there? For the outfit manager?

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/contractor/code/datums/midround/outfit.dm (hunk 6 above)
Comment (Zonespace27) [reply modular_skyrat/modules/contractor/code/datums/midround/outfit.dm:40]: Outfit manager stuff and just if I want to spawn a fully upgraded contractor without having to go through pains when I can just use "select equipment"

File: modular_skyrat/modules/contractor/code/datums/contract.dm (hunk 1 above)
Comment (Zonespace27) [reply modular_skyrat/modules/contractor/code/datums/contract.dm:188]: Added to all of them

**Review by GoldenAlpharex — APPROVED** [review]
Looks good to me.

**Pull request merged**